src/main/java/com/avioconsulting/mule/logger/internal/utils/PayloadTransformer.java

```diff
@@ -0,0 +1,139 @@
+package com.avioconsulting.mule.logger.internal.utils;
+
+import com.avioconsulting.mule.logger.api.processor.Compressor;
+import com.avioconsulting.mule.logger.api.processor.EncryptionAlgorithm;
+import com.avioconsulting.mule.logger.internal.config.CustomLoggerConfiguration;
+import com.mulesoft.modules.cryptography.api.jce.config.JceEncryptionPbeAlgorithm;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import org.mule.extension.compression.api.strategy.CompressorStrategy;
+import org.mule.extension.compression.api.strategy.gzip.GzipCompressorStrategy;
+import org.mule.runtime.api.streaming.CursorProvider;
+import org.mule.runtime.api.streaming.bytes.CursorStream;
+import org.mule.runtime.core.api.util.Base64;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.runtime.extension.api.client.DefaultOperationParameters;
+import org.mule.runtime.extension.api.client.DefaultOperationParametersBuilder;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.streaming.StreamingHelper;
+
+public class PayloadTransformer {
+
+  /**
+   * Convert payload object to Byte Array when payload resolves to a
+   * CursorProvider instance
+   * otherwise, return null
+   *
+   * @since 2.1.0
+   * @return byte array or null
+   */
+  public byte[] convertToByteArray(Object payload, StreamingHelper streamingHelper) {
+    Object resolved = streamingHelper.resolveCursorProvider(payload);
+    if (resolved instanceof CursorProvider) {
+      CursorStream cursorStream = ((CursorProvider<CursorStream>) resolved).openCursor();
+      return IOUtils.toByteArray(cursorStream);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Make transformations to payload based on logger configuration
+   * Will returned the transformed string value
+   *
+   * @since 2.1.0
+   * @return String
+   */
+  public String transformPayload(CustomLoggerConfiguration loggerConfig,
+      StreamingHelper streamingHelper, String payloadString) {
+    Objects.requireNonNull(streamingHelper, "StreamingHelper cannot be null");
+    if (payloadString == null)
+      return payloadString;
+    Compressor compressor = loggerConfig.getCompressor();
+    Result<InputStream, Void> executeCompress = null;
+    if (compressor != null) {
+      executeCompress = compressPayload(loggerConfig, payloadString);
+      try {
+        payloadString = Base64.encodeBytes(convertToByteArray(executeCompress.getOutput(), streamingHelper));
```

Comment:
Maybe we can defer this encoding to be done after encryption. Looks like `payloadString` isn't used in `encryptPayload` call if `executeCompress` is provided. Which means this is an unused transformation in that case.

The `encryptPayload` method is doing this same transformation at the end. We could return Result object from encryptPayload method and then do the transformation once at the end of this method. The Result variable could be the same carried from compress method, so whatever is final result, we encode that just once.